aws/v4/aws_v4.go

```diff
@@ -71,11 +72,21 @@ func (st Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 		return nil, err
 	}
 	resp, err := st.client.Do(req)
+
+	//Make sure resp.Body is closed, if it exists, regardless of error state
+	if resp != nil && resp.Body != nil {
+		defer resp.Body.Close()
+	}
+
 	if err != nil {
 		return nil, err
 	}
+
+	if resp == nil {
+		return nil, fmt.Errorf("Network response was empty")
```

Comment:
I do not agree with returning an error here. We should return what `st.Client.Do(...)` returns.

Reply:
I concur, I can change that